test/server/proxy-test.js

```diff
@@ -4116,5 +4081,101 @@ describe('Proxy', () => {
                 .then(() => new Promise(resolve => setTimeout(resolve, 2000)))
                 .then(() => expect(hasPageError).to.be.false);
         });
+
+        describe('Should respond with an error when destination server emits an error', () => {
```

Comment:
```js
let storedHttpRequest = null;

beforeEach(() => {
    storedHttpRequest = http.request;
});

afterEach(() => {
  http.request = storedHttpRequest;
});

```

Also, it allowing to change the `mockRequest` function signature to `function mockRequest (url, error)`